### PR TITLE
1382 - Avatar color schema to match user's schedules

### PR DESCRIPTION
**Pull request by teodosii**

### CODE REVIEW

File: grafana-plugin/src/components/ScheduleBorderedAvatar/ScheduleBorderedAvatar.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        <div className={cx('icon')}>{renderIcon()}</div>
+      </>
+    );
+  }
+
+  function renderSVG() {
+    return (
+      <>
+        <svg xmlns="http://www.w3.org/2000/svg" version="1.1" width={width} height={height} viewBox="-10 -10 220 220">
+          <g fill="none" strokeWidth="35" transform="translate(100,100)">
```
Comment (maskin25) [inline grafana-plugin/src/components/ScheduleBorderedAvatar/ScheduleBorderedAvatar.tsx:37]: hmm, why exactly these numbers `-10 -10 220 220`? Can you please declare some number constants and use them. For example :
```
const d = `M {lastX.toFixed(3)},{lastY.toFixed(3)} A {radius},{radius} 0 0,1 {x.toFixed(3)},{y.toFixed(3)}`;
```

I'm not sure is it correct in this case, I mean general idea - use understandable constants in math expressions

File: grafana-plugin/src/components/ScheduleBorderedAvatar/ScheduleBorderedAvatar.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  function renderColorPaths(colors: string[]) {
+    const colorSchemeList = colors;
+    if (colors.length === 1) { // minimum is 2 arcs to round the circle
+      colorSchemeList.push(colors[0]);
+    }
+
+    const stepAngle = (2 * Math.PI) / colors.length;
+    const radius = 100;
```
Comment (maskin25) [inline grafana-plugin/src/components/ScheduleBorderedAvatar/ScheduleBorderedAvatar.tsx:53]: use CAPITAL case for such constants please:

`const RADIUS = 100;`

File: grafana-plugin/src/containers/UsersTimezones/UsersTimezones.tsx [hunk 3]
```diff
@@ -177,6 +195,7 @@ interface AvatarGroupProps {
   xPos: number;
   currentMoment: dayjs.Dayjs;
   utcOffset: number;
+  scheduleId: string;
```
Comment (maskin25) [inline grafana-plugin/src/containers/UsersTimezones/UsersTimezones.tsx:198]: `scheduleId: Schedule['id'];`

File: grafana-plugin/src/containers/UsersTimezones/UsersTimezones.tsx [hunk 4]
```diff
@@ -277,4 +305,45 @@ const AvatarGroup = (props: AvatarGroupProps) => {
   );
 };
 
+function getColorSchemeMappingForUsers(store: RootStore, scheduleId: string): { [userId: string]: Set<string> } {
+  const startMoment = getStartOfWeek(store.currentTimezone);
+
+  const shifts: Array<{ shiftId: string; events: Event[] }> = false
```
Comment (maskin25) [inline grafana-plugin/src/containers/UsersTimezones/UsersTimezones.tsx:311]: you don't need to check `false`

File: grafana-plugin/src/pages/schedule/Schedule.tsx [hunk 5]
```diff
@@ -64,49 +66,48 @@ class SchedulePage extends React.Component<SchedulePageProps, SchedulePageState>
...
-    const users = store.userStore.getSearchResult().results;
-
+    const { query: { id: scheduleId }, store } = this.props;
+    const { isLoading, startMoment, shiftIdToShowRotationForm, shiftIdToShowOverridesForm } = this.state;
     const { scheduleStore, currentTimezone } = store;
 
+    const users = store.userStore.getSearchResult().results;
     const schedule = scheduleStore.items[scheduleId];
 
+    if (isLoading) {return <LoadingPlaceholder text="Loading..." />}
```
Comment (maskin25) [inline grafana-plugin/src/pages/schedule/Schedule.tsx:109]: remove this please, we will render some more cute loading state

File: grafana-plugin/src/pages/schedule/Schedule.tsx [hunk 6]
```diff
@@ -190,44 +166,15 @@ class SchedulePage extends React.Component<SchedulePageProps, SchedulePageState>
                   {startMoment.format('DD MMM')} - {startMoment.add(6, 'day').format('DD MMM')}
                 </div>
               </HorizontalGroup>
-              {/*<HorizontalGroup width="auto">
```
Comment (maskin25) [inline grafana-plugin/src/pages/schedule/Schedule.tsx:193]: don't remove those my comments please, I'll uncomment and return them pretty soon

File: grafana-plugin/src/models/schedule/schedule.ts [hunk 7]
```diff
@@ -76,6 +76,9 @@ export class ScheduleStore extends BaseStore {
   @observable
   byDayOptions: SelectOption[];
 
+  @observable
+  scheduleId: string;
```
Comment (maskin25) [inline grafana-plugin/src/models/schedule/schedule.ts:80]: did not get why are you storing scheduleId in schedule store, you can get schedule id in such way
```
 const { query: { id: scheduleId } } = this.props;
```
within  Schedule page

File: grafana-plugin/src/pages/schedule/Schedule.tsx [hunk 8]
```diff
@@ -64,49 +66,48 @@ class SchedulePage extends React.Component<SchedulePageProps, SchedulePageState>
...
+      store.userStore.updateItems(),
+
+      store.scheduleStore.updateFrequencyOptions(),
+      store.scheduleStore.updateDaysOptions(),
+
+      store.scheduleStore.updateOncallShifts(id), // TODO we should know shifts to render Rotations
+      this.updateEvents(),
+    ]);
+
+    store.scheduleStore.scheduleId = id;
```
Comment (maskin25) [inline grafana-plugin/src/pages/schedule/Schedule.tsx:89]: you don't need to store scheduleId in the store

File: grafana-plugin/src/pages/schedule/Schedule.tsx [hunk 9]
```diff
@@ -64,49 +66,48 @@ class SchedulePage extends React.Component<SchedulePageProps, SchedulePageState>
...
 
-    this.updateEvents();
+    this.setState({ isLoading: false });
   }
 
   componentWillUnmount() {
     const { store } = this.props;
 
     store.scheduleStore.clearPreview();
+    store.scheduleStore.scheduleId = undefined
```
Comment (maskin25) [inline grafana-plugin/src/pages/schedule/Schedule.tsx:98]: you don't need to store and erase scheduleId

**Review by maskin25 — CHANGES_REQUESTED** [review]

File: grafana-plugin/src/containers/UsersTimezones/UsersTimezones.tsx (hunk 4 above)
Comment (teodosii) [reply grafana-plugin/src/containers/UsersTimezones/UsersTimezones.tsx:311]: ahhh, forgotten. It was for debugging, good catch, I missed it.

File: grafana-plugin/src/models/schedule/schedule.ts (hunk 7 above)
Comment (teodosii) [reply grafana-plugin/src/models/schedule/schedule.ts:80]: `query` is not available on most of the `this.props` except for `Schedules` main page, previously there was some prop drilling on that - `Rotations`, `ScheduleFinal`, `ScheduleOverrides`. Why pass it down to every descendant when you can access it instead from the store?

**Review by maskin25 — APPROVED** [review]

**Pull request merged**